### ISSUE TITLE
Improving performance of parallel_stereo

### ISSUE THREAD

**Issue by oleg-alexandrov** [NeoGeographyToolkit/StereoPipeline#122]
After some fixes a while ago, parallel_stereo now correctly load-balances (as many threads overall are being used as there are cores, and if one process exits another takes its place). 

However, I noticed that stereo_tri can perform a lot better if started with bigger tiles than what is used for stereo_corr. Presumably that's because starting fewer processes with more work per process can reduce the total I/O. However, for stereo_corr we want small tiles, as that one is not I/O intensive but very CPU intensive, so bigger tiles would make it too coarse-grained. 

Also, for ISIS we may still want smaller tiles for stereo_tri as then stereo_tri is single-threaded. But even there, 2 processes with bigger tiles may perform better than 4 processes with smaller tiles, again because of I/O. 

Should we use variable-size tiles? 

Also, all processes started by parallel_stereo write to disk full-sized images, though they have only no-data outside the current tile being processed. We assumed overhead because of that would be small since LZW compresses the empty areas well. But we never tested if this scales. Does 500 processes writing to disk to 500 files bog the system down even though very little writing happens to each file most of the time?

In short, a careful analysis of parallel_stereo performance is required, both on Pleiades (which has distributed storage, and on a local machine, such as lunokhod1).

**Comment by oleg-alexandrov** [comment]
I studied this in more detail. Apparently stereo_tri and even stereo_rfne are not so good at taking advantage of using many threads. What gives the most performance is to use as many processes as there are cores and with one thread per process (for the corr, rfne, and tri steps). This is now the default. The user can always set --processess and --threads-multi if desired to tweak these.
